binary_transparency/firmware/api/annotation.go

```diff
@@ -0,0 +1,53 @@
+// Copyright 2021 Google LLC. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+// SignedStatement is a Statement signed by the Claimant.
+type SignedStatement struct {
+	// The serialised Claim in json form.
+	// This is one of MalwareStatement or BuildStatement.
+	// TODO(mhutchinson): Add an enum for Statement type?
+	Statement []byte
+
+	// Signature is the bytestream of the signature over the Statement.
+	Signature []byte
+}
+
+// FirmwareID is a pointer to a firmware version.
+// TODO(mhutchinson): This could be simplified to just LeafHash or extended to have Revision.
+type FirmwareID struct {
+	LogIndex int64
+	LeafHash []byte
+}
+
+// MalwareStatement is an annotation about malware checks in a firmware version.
+type MalwareStatement struct {
```

Comment:
Not strictly API, but need to think/note down that we will need some sort of leaf-type discriminator (enum, or ...) - currently serialised forms of these two annotations would parse as each other in some formats (e.g. protobuf- I know it says JSON up top in the definition of `Statement`, but ...)

Reply:
That's what the TODO on L21 is about: `// TODO(mhutchinson): Add an enum for Statement type?`. We can discuss it in this thread though :-)

Reply:
I've added a byte as a discriminator to the SignedStatement. I think that's reasonable for now?